src/main/java/org/jenkinsci/lib/configprovider/ConfigProvider.java

```diff
@@ -190,4 +195,21 @@ public void save(Config config) {
         GlobalConfigFiles.get().save(config);
         this.save();
     }
-}
+
+    /**
+     * Provide the given content file.
+     *
+     * @param configFile the file content to be provided
+     * @param workDir target workspace directory
+     * @param listener the listener
+     * @param tempFiles temp files created by this method, these files will
+     *                  be deleted by the caller
+     * @return file content
+     * @throws IOException in case an exception occurs when providing the content or other needed files
+     * @since 2.16
+     */
+    public String supplyContent(@NonNull Config configFile, Run<?, ?> build, FilePath workDir, TaskListener listener, @NonNull List<String> tempFiles) throws IOException {
```

Comment:
I think it would be nice to find an other solution for the `tempFiles` - It kind of looks strange to have them here in the interface. But I don't have a much better idea...